metadoc-js/src/main/scala/metadoc/ScalaDefinitionProvider.scala

```diff
@@ -0,0 +1,45 @@
+package metadoc
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import org.scalajs.dom
+import scala.meta._
+import metadoc.schema.{Index, Symbol}
+
+@ScalaJSDefined
+class ScalaDefinitionProvider(attrs: Attributes, index: Index) extends MonacoLanguages.DefinitionProvider {
+  override def provideDefinition(
+    model: MonacoEditor.IReadOnlyModel,
+    position: Monaco.Position,
+    token: Monaco.CancellationToken): MonacoLanguages.Definition = {
+
+    val offset = model.getOffsetAt(position)
+
+    val symbol: Option[String] = attrs.names.collectFirst {
+      case (pos, sym) if pos.start.offset <= offset && offset <= pos.end.offset =>
+        sym.syntax
+    }
+    val maybeDefinition = symbol.flatMap { sym =>
+      index.symbols.collectFirst {
+        case Symbol(name, Some(definition), _) if sym == name =>
+          definition
+      }
+    }
+
+    maybeDefinition match {
+      case Some(definition) =>
+        val startPos = model.getPositionAt(definition.start)
+        val endPos = model.getPositionAt(definition.end)
+
+        // FIXME: check definition.filename and reload
```

Comment:
That is a question. I didn't find an API to load from URL. The editor however has a way to change the content of the buffer, but it would be nice to have that work automatically somehow.